ui/hooks/src/use-login.new.ts

```diff
@@ -40,14 +40,16 @@ export function useConnectWallet(provider: EthProviders) {
  * ```
  */
 export function useGetLogin(onError?: (error: Error) => void) {
-  const [loginData, setLoginData] = useState(null);
+  const [loginData, setLoginData] = useState<CurrentUser>(null);
+  const [isLoading, setIsLoading] = useState(true);
 
   // check previous session
   useEffect(() => {
     const getCurrentUser = async () => {
       const sdk = getSDK();
       const data = await sdk.api.auth.getCurrentUser();
```

Comment:
It might be great to add some error handling here, try/catch at least to avoid error propagation